src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -119,8 +136,41 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         quiet.set(false)
         pull.set(false)
         cacheFrom.empty()
+        imageIdFile.set(new File(project.buildDir, ".docker/${path}-imageId.txt"))
+
+        Boolean upToDate = false;
+        if(imageIdFile.get().asFile.exists()){
+            String imageId = imageIdFile.get().asFile.text
+            
+            println "Looking in repository for imageId: $imageId"
+            DockerClient dockerClient = getDockerClient(createDockerClientConfig(), (DockerExtension) project.extensions.getByName(DockerRemoteApiPlugin.EXTENSION_NAME))
+            List<Image> images = dockerClient.listImagesCmd().exec()
+            for(image in images) {
```

Comment:
Groovy's `any` method would likely be the better option to cut down on code here.